src/replication.c

```diff
@@ -440,7 +440,13 @@ void replicationFeedSlaves(list *slaves, int dictid, robj **argv, int argc) {
 
     /* If there aren't slaves, and there is no backlog buffer to populate,
      * we can return ASAP. */
-    if (server.repl_backlog == NULL && listLength(slaves) == 0) return;
+    if (server.repl_backlog == NULL && listLength(slaves) == 0) {
+        /* We increment the repl_offset anyway, since we use that for tracking AOF fsyncs
+         * even when there's no replication active. This code will not be reached if AOF
+         * is also disabled. */
```

Comment:
This change raises two concerns:
1. Potential breakage due to replication offset incrementing continously. I can't think of a specific case but I'm not sure it implies there isn't one.
2. Potential confusion due to this behavior only limited to servers with AOF enabled.

Reply:
we can eliminate 2 by having it increment even if AOF is disabled, but it means some unnecessary overhead (we currently don't reach this code in that case).

i also think there might already be confusion of users thinking that replication offset always increments (even when the replicas drop), because we only drop the backlog after an hour, so people may not notice that and think replication offset always increment.

i understand the worry from the unknown, but i think we should proceed.
the alternative was to track two separate variables, and it greatly complicates the logic that has to work with them later.

Reply:
I can't think of anything really concerning here. I'm okay with proceeding with this.